Src/FluentAssertions/Formatting/Anchor.cs

```diff
@@ -0,0 +1,86 @@
+namespace FluentAssertions.Formatting;
+
+/// <summary>
+/// Represents a point in the formatted object graph where a new fragment or line can be inserted.
+/// </summary>
+internal class Anchor
+{
+    private readonly FormattedObjectGraph parent;
+    private readonly int indentation;
+    private readonly int characterIndex;
+    private readonly Line line;
+    private readonly bool lineWasEmptyAtCreation;
+
+    public Anchor(FormattedObjectGraph parent, Line line)
+    {
+        indentation = parent.Indentation;
+        this.parent = parent;
+        this.line = line;
+        lineWasEmptyAtCreation = line is null || line.Length == 0;
+
+        // Track the point in the graph where this instance was created.
+        characterIndex = line?.LengthWithoutOffset ?? 0;
+    }
+
+    public bool UseLineBreaks { get; set; }
+
+    public void InsertFragment(string fragment)
+    {
+        // Insert the fragment to the line and character position the anchori points at.
+        line.Insert(characterIndex, fragment);
```

Comment:
If the `Anchor` is constructed with a null `line`, this lines throws an NRE.

Reply:
Same reason as https://github.com/fluentassertions/fluentassertions/pull/3009/files/1a863c411cb9766887d81eb351ff1cc30dd72065#diff-997f82ba4a668875b0aac244469b9aec169ad8cec63ef7c0dbbc1a843008a861